iscsidev/iscsi.go

```diff
@@ -81,6 +81,10 @@ func (dev *Device) CreateTarget() (err error) {
 	if err := iscsi.AddLun(dev.targetID, TargetLunID, dev.BackingFile, dev.BSType, dev.BSOpts); err != nil {
 		return err
 	}
+	// Cannot modify the parameters for the LUNs during the adding stage
+	if err := iscsi.SetLunThinProvisioning(dev.targetID, TargetLunID); err != nil {
```

Comment:
Just a note about what fs on host will be able to supported.

-----
Not all filesystems support FALLOC_FL_PUNCH_HOLE; if a filesystem
doesn't support the operation, an error is returned.  The
operation is supported on at least the following filesystems:

       *  XFS (since Linux 2.6.38)
       *  ext4 (since Linux 3.0)
       *  Btrfs (since Linux 3.7)
       *  tmpfs(5) (since Linux 3.5)
       *  gfs2(5) (since Linux 4.16)

ref: https://man7.org/linux/man-pages/man2/fallocate.2.html

Reply:
This is a good addition for an code comment, for the `SetLunThinProvisioning` function.

Reply:
> I'm thinking if we can fallback to a non-TRIM support device when it does not support FALLOC_FL_PUNCH_HOLE.

Sounds good to me, so I would expect if non-TRIM cases will just work as today, correct?

Reply:
This thin-provisioning feature is mainly for iSCSI part. Do we mean that a filesystem **in** such kinds of thin-provisioned iSCSI device should support `FALLOC_FL_PUNCH_HOLE `?
As for the underlying filesystem of the host disks, supporting `FALLOC_FL_PUNCH_HOLE` or extents is the basic requirement, should we do anything extra for it?

Reply:
Ah, you are right. We don't do anything here.